src/eval/cache/mod.rs

```diff
@@ -0,0 +1,56 @@
+use super::{Closure, Environment, IdentKind};
+use crate::{
+    identifier::Ident,
+    term::{record::FieldDeps, BindingType, RichTerm},
+};
+
+pub mod lazy;
+//pub mod incremental;
+
+/// An index to a specific item stored in the cache
+pub type CacheIndex = lazy::Thunk;
+//pub type CacheIndex = usize;
+
+/// The current Cache implementation
+pub type CacheImpl = lazy::CBNCache;
+//pub type CacheImpl = incremental::IncCache;
+
+/// A black-holed thunk was accessed, which would lead to infinite recursion.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct BlackholedError;
+
+pub trait Cache: Clone {
```

Comment:
I think we need to document those public trait items. Could take inspiration from the corresponding function in `Thunk`, or be more concise for the time being, but otherwise I fear we're going to forget why we needed them in the first place and what they are for.

Reply:
Right. I hadn't documented them yet because I wasn't sure whether the interface was still not 100% definitive. Now I'm pretty confident there aren't going to be lots of major changes so I'll document the functions.